domain/camera/src/main/java/com/google/jetpackcamera/domain/camera/CameraXCameraUseCase.kt

```diff
@@ -228,15 +246,58 @@ constructor(
         }
     }
 
-    override fun setFlashMode(flashMode: FlashMode) {
+    override fun getScreenFlashEvents() = screenFlashEvents.asSharedFlow()
```

Comment:
> takePicture is called before the UI has subscribed

Since we are subscribing when viewmodel is initialized, I guess this case is not possible? Even if a UI part like component isn't ready or recreated, the last state based on an event should be present in `screenFlashUiState: StateFlow<ScreenFlashUiState>` of `ScreenFlash` instance in viewmodel.

> no subscribers collecting this flow (like if we unsubscribe when navigating away from the screen)

I guess we should be calling `ImageCapture#setScreenFlashUiControl(null)` while moving away since the app no longer has a valid view for a successful screen flash capture.

```
LaunchedEffect(lifecycleOwner) {
    val surfaceProvider = deferredSurfaceProvider.await()
    lifecycleOwner.repeatOnLifecycle(Lifecycle.State.STARTED) {
        viewModel.runCamera(surfaceProvider)
        try {
            awaitCancellation()
        } finally {
            viewModel.stopCamera()
        }
    }
}
```
I am assuming the right place for this should be the finally block. Initially I thought we could add it `stopCamera`, but that seems to be called for a lot of other cases as well, so maybe we can add a `PreviewViewModel#clear` method? (or `PreviewViewModel#terminate` if clear seems confusing due to `ViewModel#onCleared` which is tied to viewmodel lifecycle, not the composable lifecycle)

Reply:
```
fun stopCamera() {
    Log.d(TAG, "stopCamera")
    stopCameraJob()
    cameraUseCase.clearScreenFlash()
}

fun stopCameraJob() {
    Log.d(TAG, "stopCameraJob")
    runningCameraJob?.apply {
        if (isActive) {
            cancel()
        }
    }
}
```
Or just something like this.

Reply:
Added code to handle this in the latest commit. PTAL.

Reply:
TBH, I am now thinking that this probably shouldn't be needed in app side since CameraX lifecycle itself is being closed in this case (it's another case if CameraX lifecycle is alive but just the view in app is changing). But I am also wondering about if CameraX does handle this properly. I'll be looking into this more.

Reply:
> I guess we should be calling ImageCapture#setScreenFlashUiControl(null) while moving away since the app no longer has a valid view for a successful screen flash capture.

So this would be needed only if the view was becoming invalid while CameraX was still being used. Since JCA stops the camera and destroys the lifecycle while moving away from the view, we don't really need to do anything here and CameraX should be handling everything.

I think it's better to remove the previous code I added to address this since this should be CameraX responsibility, not app-side responsibility.

Reply:
Thanks for looking into this. It's good to hear that CameraX handles this correctly when the lifecycle is destroyed. Is it worth adding any documentation to the CameraX APIs for apps that might want to ensure all image capture requests complete before they attempt to stop the camera (while also navigating away from the view)? It might not be worth mentioning since this is likely an edge case.

Reply:
Ultimately, it ties down to ImageCapture in CameraX aborting all capture requests and notifying that to user with `onFailure(Exception("Camera closed"))`, which is not well mentioned in documentation (and may not be worth it due to being edge case, as you said, but still can be confusing to users as it is considered a failure with `Exception` as error).

We will need to discuss this more with the team.